### DISCUSSION TITLE
Question about usage

### DISCUSSION THREAD

**Discussion by gbtorrance** [eprbell/rp2#91] [General]
Hi. Working on my taxes with the help of RP2. (Thank you!)

I have a few questions about the intended usage for RP2 from year to year. Maybe dumb questions, but I am a bit confused.

Is the intent that one will just continue appending new transactions to the same ODS input file year after year, or create a new ODS file for each year?

What happens once there are multiple tax years' worth of data in the ODS? Will RP2 only generate tax report information for the most recent tax year, or for all tax years for which it has information?

And how do the `--from-date` and `--to-date` options work? `--to-date` seems fairly obvious, but will `--from-date` completely ignore transaction information from prior to that date, or just not report on it? (I assume the latter, because otherwise the results would be all wrong.)

Thanks!

(BTW, I still intend to work on the Binance.US REST plugin. It will take a while, as I'm planning to work on it during my work "hacker hours", as allowed by my company. Currently don't have much time outside of that to dedicate to it.)

**Comment by eprbell** [comment]
Good questions! See inline replies below.

> Is the intent that one will just continue appending new transactions to the same ODS input file year after year, or create a new ODS file for each year?

Yes, just continue appending to the same .ods file. The input that is given to RP2 must always contain the full transaction history from the beginning.

> What happens once there are multiple tax years' worth of data in the ODS? Will RP2 only generate tax report information for the most recent tax year, or for all tax years for which it has information?

Both the rp2_full_report and the tax_report plugins generate the full history by default. However their output can be filtered, using `--from-date` and `--to-date`. These two options don't change anything in the internal calculations of the RP2 tax engine: they only allow users to select what ends up in the output. These options are equivalent to generating the full file and then opening it in the spreadsheet program and manually deleting all transactions before the "from date" and after the "to date".

> And how do the `--from-date` and `--to-date` options work? `--to-date` seems fairly obvious, but will `--from-date` completely ignore transaction information from prior to that date, or just not report on it? (I assume the latter, because otherwise the results would be all wrong.)

No, `--from-date` and `--to-date` are only output filters: they don't affect internal representation and calculations. So `--from-date` is used to skip transactions before the "from date" in the output (but those transactions would have been calculated anyway by RP2 and would just be ignored by the output generators).

> (BTW, I still intend to work on the Binance.US REST plugin. It will take a while, as I'm planning to work on it during my work "hacker hours", as allowed by my company. Currently don't have much time outside of that to dedicate to it.)

That's great! Take your time: we're all volunteers and busy with other stuff too :-)

**Reply by kka54ru** [reply]
@eprbell but for huge volumes of transactions such approach will be prone to human errors in dealing with many rows and take more and more memory and time each year. I reckon being able to use open positions report as a base for new transaction file processing is a good feature for this software. 
Do you see any conceptual/architectural issues with that?

Thank you for your effort!

**Comment by gbtorrance** [comment]
Perfect! Thanks for the reply and the info!
